spec/models/miq_request_workflow_spec.rb

```diff
@@ -350,19 +350,40 @@
         end
 
         context "with a user with a belongs_to_filter" do
+          let(:root_folder) { FactoryBot.create(:ems_folder, :ems_id => ems.id).tap        { |f| f.add_parent(ems) } }
+          let(:datacenter)  { FactoryBot.create(:vmware_datacenter, :ems_id => ems.id).tap { |d| d.add_parent(root_folder) } }
+          let(:storage)     { FactoryBot.create(:storage, :ems_id => ems.id).tap           { |s| s.add_parent(datacenter) } }
```

Comment:
NOTE these add_parent calls are the fix for the belongsto_filter not working as expected before, the filter had to match the ems_metadata tree otherwise miq_filter would return [] and bypass the preloader

Reply:
Thanks, I wondered what that was for.

Reply:
Yeah specifically the issue was https://github.com/ManageIQ/manageiq/blob/eccb74912c9d63b85b6a5c9e2c8289015df6debe/app/models/miq_filter.rb#L71 since the "leaf" wasn't at the expected depth, the filter was ems/folder/datacenter so it had depth 3 but the datacenter had no parents so it was at depth 0.  This meant that https://github.com/ManageIQ/manageiq/blob/master/lib/rbac/filterer.rb#L916 would lead to `sources` being empty so the preloader https://github.com/ManageIQ/manageiq/blob/master/lib/rbac/filterer.rb#L918 didn't hit the missing virtual_has_many

Reply:
wow